cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java

```diff
@@ -150,246 +154,253 @@ public void onFailure(int statusCode, String response, Throwable throwable) {
+            scheduleBanners();
+        });
     }
 
     private void createBanners(Collection<Banner> banners) {
-		for (Banner banner : banners) {
+        for (Banner banner : banners) {
             if (banner.getStatus() == BannerStatus.Draft && !showDrafts) {
-            	Log.d(TAG, "Skipping Banner because: Draft");
+                Log.d(TAG, "Skipping Banner because: Draft");
                 continue;
             }
 
             if (banner.getFrequency() == BannerFrequency.Once && isBannerShown(banner.getId())) {
-				Log.d(TAG, "Skipping Banner because: Frequency");
-                continue;
-            }
-
-            Date now = new Date();
-            if (banner.getStopAtType() == BannerStopAtType.SpecificTime && banner.getStopAt().after(now)) {
-				Log.d(TAG, "Skipping Banner because: Time");
+                Log.d(TAG, "Skipping Banner because: Frequency");
                 continue;
             }
 
             if (banner.getTriggerType() == BannerTriggerType.Conditions) {
-            	boolean triggers = false;
-            	for (BannerTrigger trigger : banner.getTriggers()) {
-            		boolean triggerTrue = false;
-            		for (BannerTriggerCondition condition : trigger.getConditions()) {
-            			boolean conditionTrue = false;
-            			if (condition.getType() != null) {
-							if (condition.getType().equals(BannerTriggerConditionType.Duration)) {
-								banner.setDelaySeconds(condition.getSeconds());
-								conditionTrue = true;
-							}
-							if (condition.getType().equals(BannerTriggerConditionType.Sessions)) {
-								if (condition.getRelation().equals("lt")) {
-									conditionTrue = sessions < condition.getSessions();
-								} else {
-									conditionTrue = sessions > condition.getSessions();
-								}
-							}
-							if (condition.getType().equals(BannerTriggerConditionType.Event)) {
-								String event = events.get(condition.getKey());
-								conditionTrue = event != null && event.equals(condition.getValue());
-							}
-						}
-
-						if (conditionTrue) {
-							triggerTrue = true;
-							break;
-						}
-					}
-
-            		if (triggerTrue) {
-						triggers = true;
-						break;
-					}
-				}
-
-            	if (!triggers) {
-					Log.d(TAG, "Skipping Banner because: Trigger not satisfied " + sessions);
-            		continue;
-				}
-			}
+                boolean triggers = false;
+                for (BannerTrigger trigger : banner.getTriggers()) {
+                    boolean triggerTrue = false;
+                    for (BannerTriggerCondition condition : trigger.getConditions()) {
+                        boolean conditionTrue = false;
+                        if (condition.getType() != null) {
+                            if (condition.getType().equals(BannerTriggerConditionType.Duration)) {
+                                banner.setDelaySeconds(condition.getSeconds());
+                                conditionTrue = true;
```

Comment:
please do not abuse booleans like this. find a proper statement to terminate the for loop please